service/scaladsl/kafka/server/src/main/scala/com/lightbend/lagom/internal/scaladsl/broker/kafka/ScaladslRegisterTopicProducers.scala

```diff
@@ -56,27 +59,30 @@ class ScaladslRegisterTopicProducers(
               case tagged: TaggedOffsetTopicProducer[Any, _] =>
                 val tags = tagged.tags
 
-                val eventStreamFactory: (String, Offset) => Source[(Any, Offset), _] = { (tag, offset) =>
+                val eventStreamFactory: (String, Offset) => Source[(Message[Any], Offset), _] = { (tag, offset) =>
                   tags.find(_.tag == tag) match {
                     case Some(aggregateTag) => tagged.readSideStream(aggregateTag, offset)
                     case None               => throw new RuntimeException("Unknown tag: " + tag)
                   }
                 }
 
-                val partitionKeyStrategy: Option[Any => String] = {
-                  topicCall.properties.get(KafkaProperties.partitionKeyStrategy).map { pks => message =>
-                    pks.computePartitionKey(message)
+                val transform: Message[Any] => ProducerRecord[String, Any] = message => {
+                  val key: String = topicCall.properties.get(KafkaProperties.partitionKeyStrategy) match {
+                    case Some(strategy) => strategy.computePartitionKey(message.payload)
+                    case None           => null
```

Comment:
The same could have been written as:

```scala
val key: String = 
  topicCall.properties.get(KafkaProperties.partitionKeyStrategy)
	.map(_.computePartitionKey(message.payload))
    .orNull
```

Matter of taste, no big issue. It just aligns with the style used below, in line 74.

Reply:
adjusted